src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee

```diff
@@ -0,0 +1,43 @@
+@App.component('mnoeImpersonateButton', {
+  templateUrl: 'app/components/mnoe-impersonate-button/mnoe-impersonate-button.html',
+  bindings: {
+    organizationId: '<'
+    wrapperClassNames: '@'
+    btnClassNames: '@'
+    classNames: '@',
+    user: '<',
+  }
+  controllerAs: 'vm'
+  controller: ($window, toastr, MnoErrorsHandler, MnoeAdminConfig, MnoeUsers) ->
+    vm = this
+    vm.isImpersonationEnabled = MnoeAdminConfig.isImpersonationEnabled()
+
+    vm.impersonationStatus = ->
+      if MnoeAdminConfig.isImpersonationConsentRequired()
+        switch vm.user.access_request_status
+          when 'approved' then 'approved'
+          when 'never_requested', 'expired', 'denied' then 'requestable'
+          when 'requested' then 'requested'
+      else
+        'approved'
```

Comment:
For those labels, maybe we could do something like this (cf. https://github.com/alexnoox/mnoe-admin-panel/blob/1.0/src/app/index.constants.coffee):
```
.constant('USER_ROLES', [
    {value: 'Member', label: 'mnoe_admin_panel.constants.user_roles.member'},
    {value: 'Admin', label: 'mnoe_admin_panel.constants.user_roles.admin'},
    {value: 'Super Admin', label: 'mnoe_admin_panel.constants.user_roles.super_admin'}
  ])
```